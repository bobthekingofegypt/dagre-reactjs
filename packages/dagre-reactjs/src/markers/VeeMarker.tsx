import * as React from 'react';

import { MarkerProps } from '../types';

export const VeeMarker: React.FC<MarkerProps> = ({ edgeMeta, markerId }) => {
  return (
    <marker
      id={markerId}
      viewBox="0 0 10 10"
      refX="9"
      refY="5"
      markerUnits="strokeWidth"
      markerWidth="8"
      markerHeight="6"
      orient="auto"
    >
      <path
        d="M 0 0 L 10 5 L 0 10 L 4 5 z"
        className={edgeMeta.styles.marker.className}
        style={
          edgeMeta.styles.marker.styles || {
            strokeWidth: 1,
            strokeDasharray: '1, 0',
          }
        }
      ></path>
    </marker>
  );
};
