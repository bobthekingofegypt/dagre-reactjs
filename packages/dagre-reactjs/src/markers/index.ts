import { MarkerComponent, MarkerComponents } from '../types';
import { NormalMarker } from './NormalMarker';
import { UndirectedMarker } from './UndirectedMarker';
import { VeeMarker } from './VeeMarker';

export const builtInMarkers: MarkerComponents = {
  normal: NormalMarker,
  vee: VeeMarker,
  undirected: UndirectedMarker,
};

export const getMarkerComponent = (
  type: string,
  markerComponents: MarkerComponents
): MarkerComponent => {
  if (markerComponents[type]) {
    return markerComponents[type];
  }

  throw Error(`marker component ('${type}') not found in builtins or custom`);
};
