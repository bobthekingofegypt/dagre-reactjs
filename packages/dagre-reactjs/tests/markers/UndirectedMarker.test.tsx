import { shallow } from 'enzyme';
import * as React from 'react';

import { UndirectedMarker } from '../../src/markers/UndirectedMarker';
import { getEdgeMeta } from '../stubs';

describe('UndirectedMarker', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <UndirectedMarker
        edgeMeta={getEdgeMeta('1', '2', 'testlabel')}
        markerId="testID"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('sets correct id to passed id', () => {
    const edgeMeta = getEdgeMeta('1', '2', 'test label');

    const wrapper = shallow(
      <UndirectedMarker edgeMeta={edgeMeta} markerId="monkey" />
    );
    const path = wrapper.find('marker');
    expect(path.prop('id')).toBe('monkey');
  });

  it('uses marker classname if sent', () => {
    const edgeMeta = getEdgeMeta('1', '2', 'test label');
    edgeMeta.styles.marker.className = 'test-classname';

    const wrapper = shallow(
      <UndirectedMarker edgeMeta={edgeMeta} markerId="monkey" />
    );
    const path = wrapper.find('path');
    expect(path.hasClass('test-classname')).toBe(true);
  });

  it('uses custom styles if passed', () => {
    const edgeMeta = getEdgeMeta('1', '2', 'test label');
    edgeMeta.styles.marker.styles = {
      fill: '#000',
    };

    const wrapper = shallow(
      <UndirectedMarker edgeMeta={edgeMeta} markerId="monkey" />
    );
    const path = wrapper.find('path');
    expect(path.prop('style')).toEqual(edgeMeta.styles.marker.styles);
  });
});
