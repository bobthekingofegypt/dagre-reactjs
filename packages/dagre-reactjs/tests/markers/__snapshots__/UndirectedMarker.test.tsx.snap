// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UndirectedMarker renders correctly 1`] = `
<marker
  id="testID"
  markerHeight="6"
  markerUnits="strokeWidth"
  markerWidth="8"
  orient="auto"
  refX="9"
  refY="5"
  viewBox="0 0 10 10"
>
  <path
    d="M 0 5 L 10 5"
    style={
      Object {
        "strokeDasharray": "1, 0",
        "strokeWidth": 1,
      }
    }
  />
</marker>
`;
